code/game/objects/items/teleportation.dm

```diff
@@ -219,3 +219,17 @@
 		if(active_portal_pairs[i] == P)
 			return DESTINATION_PORTAL
 	return FALSE
+
+/obj/item/hand_tele/suicide_act(mob/user)
+	if(iscarbon(user))
+		user.visible_message("<span class='suicide'>[user] is creating a weak portal and sticking [user.p_their()] head through! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		var/mob/living/carbon/itemUser = user
+		var/obj/item/bodypart/head/head = itemUser.get_bodypart(BODY_ZONE_HEAD)
+		if(head)
+			head.drop_limb()
+			var/list/safeLevels = SSmapping.levels_by_trait(ZTRAIT_SPACE_RUINS) + SSmapping.levels_by_trait(ZTRAIT_LAVA_RUINS) + SSmapping.levels_by_trait(ZTRAIT_STATION) + SSmapping.levels_by_trait(ZTRAIT_MINING)

```

Comment:
It can't be made a static since things might change. It's fine as is.

Reply:
If Z-levels are changing like that then you have a problem.

Reply:
Even if that's true, I think you should have that as a global that updates when new Z-levels are Added/Removed and use that here.

Reply:
The overhead of these calls, infrequent as they are, is way less than tracking everything globally. But I would suggest to use `levels_by_any_trait` rather than `+`.

Reply:
@SpaceManiac How does it work?

Reply:
`var/list/safeLevels = SSmapping.levels_by_any_trait(list(ZTRAIT_SPACE_RUINS, ZTRAIT_LAVA_RUINS, ZTRAIT_STATION, ZTRAIT_MINING))``